### PR TITLE
Added support for runtime expressions embedded in string literals

### PR DESCRIPTION
**Pull request by darrelmiller**
Addresses issue #287

### CODE REVIEW

File: src/Microsoft.OpenApi/Expressions/CompositeExpression.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using System.Threading.Tasks;
+
+namespace Microsoft.OpenApi.Expressions
+{
+    /// <summary>
+    /// String literal with embedded expressions
+    /// </summary>
+    public class CompositeExpression : RuntimeExpression
+    {
+        private readonly string template;
```
Comment (PerthCharern) [inline src/Microsoft.OpenApi/Expressions/CompositeExpression.cs]: >template [](start = 32, length = 8)

not sure if template is the right word here... maybe "originalExpression" ?

File: src/Microsoft.OpenApi/Expressions/CompositeExpression.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        private Regex expressionPattern = new Regex("{(?<exp>[^}]+)");
+        /// <summary>
```
Comment (PerthCharern) [inline src/Microsoft.OpenApi/Expressions/CompositeExpression.cs]: nit, please add one newline for readability

File: src/Microsoft.OpenApi/Expressions/CompositeExpression.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        public CompositeExpression(string expression)
+        {
+            template = expression;
+
+            // Extract subexpressions and convert to RuntimeExpressions
+            var matches = expressionPattern.Matches(expression);
+
+            foreach (var item in matches.Cast<Match>())
+            {
+                var value = item.Groups["exp"].Captures.Cast<Capture>().First().Value;
```
Comment (PerthCharern) [inline src/Microsoft.OpenApi/Expressions/CompositeExpression.cs]: >value [](start = 20, length = 5)

nit, call this runtimeExpression instead of value

**Comment by PerthCharern** [comment]
Can we add one test using composite expression in [https://github.com/Microsoft/OpenAPI.NET/blob/master/test/Microsoft.OpenApi.Readers.Tests/V3Tests/OpenApiCallbackTests.cs](https://github.com/Microsoft/OpenAPI.NET/blob/master/test/Microsoft.OpenApi.Readers.Tests/V3Tests/OpenApiCallbackTests.cs) ?

**Comment by darrelmiller** [comment]
@PerthCharern I will make these changes.  How do feel about releasing 1.1, rather than me back porting these changes to 1.0?

**Comment by PerthCharern** [comment]
Sounds fine to me.

Perth Charernwattanagul
Sent from my mobile device

On Jul 9, 2018, at 12:13, Darrel <notifications@github.com<mailto:notifications@github.com>> wrote:


@PerthCharern<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FPerthCharern&data=02%7C01%7CPerth.Charernwattanagul%40microsoft.com%7Cc0812b22644e44f21f1d08d5e5d00599%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636667603971401302&sdata=gQ3s45EjxE2kr5qShLE6%2BVu3L%2Fb1OXVF6p8t4PPft7w%3D&reserved=0> I will make these changes. How do feel about releasing 1.1, rather than me back porting these changes to 1.0?

**Pull request merged**